File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.js
function solution(sizes) {
    var answer = 0;
    sizes.map((size)=>size.sort((a,b)=>b-a));
    let width=0;
    let heigth=0;
    for(let i = 0;i<sizes.length;i++){
        if(width<sizes[i][0])width=sizes[i][0];
        if(heigth<sizes[i][1])heigth=sizes[i][1]
    }
    answer = width*heigth;
    return answer;
}
// 가로 세로 중 더 긴걸 가로로 하는 새로운 배열 생성
// 최대 가로와 최소 가로  

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> sizes) {
    int answer = 0;
    int maxW = 0;
    int maxH = 0; 
    for(int i=0;i<sizes.size();i++){
        int w = sizes[i][0];
        int h = sizes[i][1];
        if(w < h ){
            sizes[i][0]=h;
            sizes[i][1]=w;
        }
        
        if(maxW < sizes[i][0]) maxW = sizes[i][0];
        if(maxH < sizes[i][1]) maxH = sizes[i][1];
    }
    return maxW * maxH;
}